### ISSUE TITLE
Firestore crashes since 11.23

### ISSUE THREAD

**Issue by mpsq** [firebase/firebase-tools#5523]
### Environment info

- v11.23.0 (and 11.23.1)
- Ubuntu

### Test case

Firestore emulators returns:

```
{"error":{"code":500,"message":"RST_STREAM closed stream. HTTP\/2 error code: PROTOCOL_ERROR","status":"INTERNAL"}}
```

When requests are made with the `Proxy-Connection: keep-alive` (or `Connection: keep-alive`) header. This regression was introduced in https://github.com/firebase/firebase-tools/pull/5493. The request is made with HTTP 1.1.

### Expected behavior

The Firestore emulator used to handle `Proxy-Connection` & `Connection` headers. 

### Actual behavior

Firestore emulator seems use the wrong protocol (HTTP2).

**Comment by aalej** [comment]
Hi @mpsq, thanks for reaching out to us. Is there any chance you could share an [MCVE](https://stackoverflow.com/help/minimal-reproducible-example) so that we can replicate the issue, or a code snippet and detailed steps so that will help us produce the same error message?

**Comment by mpsq** [comment]
Absolutely, run the Firestore emulator, then send the following request with cURL:

```
#!/usr/bin/env sh

curl -X POST "http://127.0.0.1:8080/v1/projects/tests/databases/(default)/documents/tests?documentId=5UmoNy6ua23qSmPBomAjboYMxTf2" \
    -H 'Authorization: Bearer owner' \
    -H 'Proxy-Connection: keep-alive'
```

**Comment by aalej** [comment]
Thanks @mpsq for the additional information, I was able to reproduce the behavior you mentioned. That said, I’ll notify our team about this so that we can determine if this behavior is intended. I'll also be marking this issue as reproducible.

**Comment by aalej** [comment]
Hi @mpsq, while we're investigating this issue, I would recommend downgrading to firebase-tools v11.22.0 if you are using any HTTPS request that are made with the header `Proxy-Connection: keep-alive`. Apologies for the inconvenience this has caused, and thanks for reporting this to us.

**Comment by christhompsongoogle** [comment]
The workaround in the short term is to use HTTPv2 as we investigate how to resolve this one

**Comment by christhompsongoogle** [comment]
We will be rolling out a change this week that should resolve the issue.

**Comment by mpsq** [comment]
I confirm that this is now fixed :+1:
